File: src/main/java/ash/java/graphql/fields/KeywordSchema.java
package ash.java.graphql.fields;

import ash.java.graphql.data.MovieDao;
import ash.java.graphql.types.keyword.KeywordType;
import graphql.schema.*;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;


import static graphql.Scalars.*;
import static graphql.schema.GraphQLFieldDefinition.newFieldDefinition;

@Service
public class KeywordSchema implements FieldProducer {

    private MovieDao movieDao;

    @Autowired
    public KeywordSchema(MovieDao movieDao) {
        this.movieDao = movieDao;
    }

    private GraphQLFieldDefinition keywordResultFieldType = newFieldDefinition()
            .type(new GraphQLList(new KeywordType().getGraphQlType()))
            .name("keywordList")
            .argument(arg -> arg.name("filmId")
                    .type(GraphQLInt))
            .dataFetcher(env -> movieDao.getKeywordsForMovie(env.getArgument("filmId")))
            .build();

    public GraphQLFieldDefinition getFieldDefinition() {
        return keywordResultFieldType;
    }
}

File: src/main/kotlin/ash/kotlin/graphql/fields/KeywordSchema.kt
package ash.kotlin.graphql.fields

import ash.kotlin.graphql.data.MovieDao
import ash.kotlin.graphql.types.keyword.KeywordType
import graphql.Scalars
import graphql.schema.GraphQLFieldDefinition
import graphql.schema.GraphQLList

class KeywordSchema constructor(private val dao: MovieDao) : FieldProducer
{
    override fun getFieldDefinition(): GraphQLFieldDefinition
    {
        return GraphQLFieldDefinition.newFieldDefinition()
                .type(GraphQLList(KeywordType().getGraphQlType()))
                .name("keywordList")
                .argument { arg -> arg.name("filmId").type(Scalars.GraphQLInt) }
                .dataFetcher { env -> dao.getKeywordsForMovie(env.getArgument("filmId")) }
                .build()
    }
}